### ISSUE TITLE
Strings with a number and whitespaces are considered numeric

### ISSUE THREAD

**Issue by vladshcherbin** [plotly/fast-isnumeric#1]
Hey, should a string with whitespaces be numeric:

```js
var isNumeric = require('fast-isnumeric')
console.log(isNumeric('   3 ')) // -> true
```

I'd expect it to be false.

**Comment by alexcjohnson** [comment]
Yes, leading and trailing whitespace is accepted - as expressed in the [tests](https://github.com/plotly/fast-isnumeric/blob/efdd602521132349539fcf32161f4f62f10ea741/test.js#L113-L115). This is fairly commonly seen when accepting user input or reading data from files of unknown format - users read these strings as numbers, so we do too.

**Comment by vladshcherbin** [comment]
@alexcjohnson why though?

For example, if I expect to receive from user a value `'33'`, he can send it as `'   33'` or `'33   '`, etc. Do you trim all input values later?

**Comment by alexcjohnson** [comment]
> Do you trim all input values later?

No, I turn them into actual numbers, once I know that that's what they represent.

The question this repo is answering (same as [is-number](https://github.com/jonschlinkert/is-number) which I see you also [commented on](https://github.com/jonschlinkert/is-number/issues/13)) is "does the object I was passed represent a number?" and I think it's pretty clear that `' 33 '` DOES represent a number, as opposed to some non-numeric text like a date string or categorical data.

It seems like you're asking a different question about your input data - perhaps "is it a properly formatted number?" or something...

**Comment by vladshcherbin** [comment]
@alexcjohnson yep, you're probably right.

I was fooled by the differences of what is considered `numeric` in different languages. I checked this in php with `is_numeric` function and the result was `false`. After your words I came back and checked [this two versions](https://repl.it/repls/WeeInfamousAsianpiedstarling) and remembered how retarded php is 🤦‍♂️

Thanks you for your time and explanation, sorry for bothering.
